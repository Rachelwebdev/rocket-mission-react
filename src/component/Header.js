import React from 'react';
import styles from '../styles/Header.module.css';
import planetImg from '../assets/planet.png';
import Navbar from './Navbar';

const Header = () => (
  <header>
    <div className={styles.navHeader}>
      <div className={styles.logoContainer}>
        <img
          src={planetImg}
          alt="planet logo"
          className="logo"
          width="50"
          height="50"
        />
        <h2>Space Travelers Hub</h2>
      </div>
      <Navbar />
    </div>
    <hr />
  </header>
);

export default Header;
