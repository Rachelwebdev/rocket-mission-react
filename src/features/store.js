import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './missions/missionSlice';
import rocketsReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
    rockets: rocketsReducer,
  },
});
export default store;
