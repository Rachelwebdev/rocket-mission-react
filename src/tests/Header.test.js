import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from '../component/Header';
import store from '../features/store';

describe('Header component', () => {
  it('should render Header component', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
