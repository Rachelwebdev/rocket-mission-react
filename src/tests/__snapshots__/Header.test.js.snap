// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<div>
  <header>
    <div
      class="navHeader"
    >
      <div
        class="logoContainer"
      >
        <img
          alt="planet logo"
          class="logo"
          height="50"
          src="planet.png"
          width="50"
        />
        <h2>
          Space Travelers Hub
        </h2>
      </div>
      <div>
        <nav
          class="navLinks"
        >
          <a
            aria-current="page"
            class="navLink active"
            href="/"
            style="text-decoration: underline;"
          >
            My Profile
          </a>
          <a
            class="navLink"
            href="/missions"
            style="text-decoration: none;"
          >
            Missions
          </a>
          <a
            class="navLink"
            href="/rockets"
            style="text-decoration: none;"
          >
            Rockets
          </a>
        </nav>
      </div>
    </div>
    <hr />
  </header>
</div>
`;
